src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -857,6 +882,8 @@ public async Task ReadResponseHeadersAsync(CancellationToken cancellationToken)
                         (wait, emptyResponse) = TryEnsureHeaders();
                         Debug.Assert(!wait);
                     }
+
+                    if (HttpTelemetry.Log.IsEnabled()) HttpTelemetry.Log.ResponseHeadersBegin();
```

Comment:
Is this happening before or after the headers are read?
I'm asking whether this is logically called at the same "time" as for HTTP/1.1.

Reply:
The idea behind this event: https://github.com/dotnet/runtime/pull/40838#discussion_r471084360

I moved it to `OnHeadersStart` for HTTP2 to move the timestamp closer to the one for HTTP1. As you pointed out, it would have been called **after** receiving all headers for Http2 before.